include/boost/json/impl/string.hpp

```diff
@@ -77,12 +79,48 @@ insert(
         first, last, dsp,
         iter_cat<InputIt>{});
     cleanup c{tmp, dsp};
-    auto const off = pos - impl_.data();
     traits_type::copy(
-        impl_.insert(off, tmp.size(), sp_),
+        impl_.insert_unchecked(pos, tmp.size(), sp_),
         tmp.data(),
         tmp.size());
-    return impl_.data() + off;
+    return *this;
+}
+
+// KRYSTIAN TODO: this can be done without copies when
+// reallocation is not needed.
```

Comment:
Fix the comment to explain that we should use a separate algorithm if iterator category is Forward or better

Reply:
Done